### PR TITLE
use native support for Galera Clustering in ProxySQL 2.0

### PR DESCRIPTION
**Pull request by delgod**

### CODE REVIEW

File: proxysql-admin [hunk 1]
```diff
@@ -1189,16 +1191,16 @@ function enable_proxysql() {
         proxysql_exec "$LINENO" "SELECT hostname,hostgroup_id,port,weight,comment FROM mysql_servers WHERE hostgroup_id=$WRITE_HOSTGROUP_ID" "-t"
         echo ""
       else
-        proxysql_exec "$LINENO" "INSERT INTO mysql_servers (hostname,hostgroup_id,port,weight,comment,max_connections) VALUES ('$ws_ip',$READ_HOSTGROUP_ID,$ws_port,1000,'READ',$MAX_CONNECTIONS);"
+        proxysql_exec "$LINENO" "INSERT INTO mysql_servers (hostname,hostgroup_id,port,weight,comment,max_connections) VALUES ('$ws_ip',$WRITE_HOSTGROUP_ID,$ws_port,1000,'READ',$MAX_CONNECTIONS);"
         check_cmd $? $LINENO "Failed to add the PXC server node $i."\
                               "\n-- Please check the ProxySQL connection parameters and status."
       fi
     done
-    if [[ $WITH_CLUSTER_APP_USER -eq 1 ]]; then
```
Comment (delgod) [inline proxysql-admin:1197]: I believe we need to add query rules for all users, not only for Cluster App User

File: proxysql-admin [hunk 2]
```diff
@@ -1110,12 +1112,12 @@ function enable_proxysql() {
     proxysql_load_to_runtime_save_to_disk "MYSQL SERVERS" $LINENO
```
Comment (delgod) [inline proxysql-admin:1112]: not sure if `MODE=loadbal` possible in native Galera Clustering

File: proxysql-admin [hunk 3]
```diff
@@ -1259,29 +1265,30 @@ function disable_proxysql(){
                        "\n-- Please check the PXC connection parameters and status."
   readonly cluster_name
 
-  echo "Removing default cluster application user from ProxySQL database."
-  proxysql_exec "$LINENO" "DELETE FROM mysql_users WHERE username='$CLUSTER_APP_USERNAME';"
-  check_cmd $? $LINENO "Failed to delete the PXC user ($CLUSTER_APP_USERNAME) from ProxySQL."\
-                       "\n-- Please check the ProxySQL connection parameters and status."
+  if [[ $WITH_CLUSTER_APP_USER -eq 1 ]]; then
```
Comment (delgod) [inline proxysql-admin:1268]: don't fail if `WITH_CLUSTER_APP_USER=0`

File: proxysql-admin [hunk 4]
```diff
@@ -1964,9 +1971,9 @@ function parse_args() {
     SLAVEREAD_HOSTGROUP_ID=$READ_HOSTGROUP_ID
   fi
 
-  if [[ ! $WRITER_IS_READER =~ ^(always|never|ondemand)$ ]]; then
+  if [[ ! $WRITER_IS_READER =~ ^(0|1|2)$ ]]; then
     error "" "Invalid --writer-is-reader option: '$WRITER_IS_READER'"
```
Comment (rameshvs02) [inline proxysql-admin:1975]: @delgod  proxysql-admin -d is failing with following error.
```
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$ sudo ./proxysql-admin -d
ERROR : Invalid --writer-is-reader option: 'ondemand'
Please choose one of these values: 0(disabled), 1(always), or 2(only backup writer is reader)
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$
```
Variable `WRITER_IS_READER` is present in default configuration file proxysql-admin.cnf. We need to update the variable there too.
```
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$ sudo grep -B2 "WRITER_IS_READER" /etc/proxysql-admin.cnf

# Writer-is-reader configuration
export WRITER_IS_READER="ondemand"
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$
``` 
bash -x outptut
```
+ SLAVEREAD_HOSTGROUP_ID=21
+ [[ ! ondemand =~ ^(0|1|2)$ ]]
+ error '' 'Invalid --writer-is-reader option: '\''ondemand'\'''
+ local lineno=
+ shift
+ [[ -n '' ]]
+ printf 'ERROR : Invalid --writer-is-reader option: '\''ondemand'\''\n'
ERROR : Invalid --writer-is-reader option: 'ondemand'
+ echo 'Please choose one of these values: 0(disabled), 1(always), or 2(only backup writer is reader)'
Please choose one of these values: 0(disabled), 1(always), or 2(only backup writer is reader)
+ exit 1
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$
```

Comment (delgod) [reply proxysql-admin:1975]: @rameshvs02, yes, WRITER_IS_READER option in ProxySQL 2.0 is not compatible. It is expected

File: proxysql-admin [hunk 5]
```diff
@@ -76,8 +76,10 @@ declare    SLAVE_IS_WRITER=""
 declare    MODE="singlewrite"
 declare -i NODE_CHECK_INTERVAL=3000
 
-declare -i WRITE_HOSTGROUP_ID=10
-declare -i READ_HOSTGROUP_ID=11
+declare -i READ_HOSTGROUP_ID=10
+declare -i WRITE_HOSTGROUP_ID=11
+declare -i BACKUP_WRITE_HOSTGROUP_ID=12
+declare -i OFFLINE_HOSTGROUP_ID=13
```
Comment (rameshvs02) [inline proxysql-admin:82]: @delgod We should add `BACKUP_WRITE_HOSTGROUP_ID` and `OFFLINE_HOSTGROUP_ID` in proxysql-admin.cnf. If we need to add two PXC cluster in proxysql db there will be hostgroup conflict. We can give different hostgroups if we have these variables in proxysql-admin.cnf

File: proxysql-admin (hunk 2 above)
Comment (rameshvs02) [reply proxysql-admin:1112]: @delgod loadbal mode will fail with the following error.
```
Application user 'cluster_two'@'127.%' already present in PXC.

Adding the Percona XtraDB Cluster server nodes to ProxySQL
ERROR 1045 (28000) at line 1: CHECK constraint failed: mysql_galera_hostgroups
ERROR (line:1238) : Failed to add the PXC monitoring scheduler in ProxySQL.
-- Please check the ProxySQL connection parameters and status.
ramesh@qaserver-05:/qa/8pxc/work/proxysql-admin-tool$
```

Comment (kennt-percona) [reply proxysql-admin:1112]: The loadbal mode is essentially (1) all nodes are writers and (2) max writers is a large number (say 10000).

**Pull request closed without merging**